ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixParticipantTest.java

```diff
@@ -63,20 +66,103 @@
 public class HelixParticipantTest {
   private final MockHelixManagerFactory helixManagerFactory;
   private final Properties props;
+  private final String clusterName = "HelixParticipantTestCluster";
 
   public HelixParticipantTest() throws Exception {
     List<com.github.ambry.utils.TestUtils.ZkInfo> zkInfoList = new ArrayList<>();
     zkInfoList.add(new com.github.ambry.utils.TestUtils.ZkInfo(null, "DC0", 2199, false));
     JSONObject zkJson = constructZkLayoutJSON(zkInfoList);
     props = new Properties();
     props.setProperty("clustermap.host.name", "localhost");
-    props.setProperty("clustermap.cluster.name", "HelixParticipantTestCluster");
+    props.setProperty("clustermap.cluster.name", clusterName);
     props.setProperty("clustermap.datacenter.name", "DC0");
     props.setProperty("clustermap.dcs.zk.connect.strings", zkJson.toString(2));
     helixManagerFactory = new MockHelixManagerFactory();
   }
 
   /**
+   * Tests setReplicaSealedState method for {@link HelixParticipant}
+   * @throws IOException
+   */
+  @Test
+  public void testSetReplicaSealedState() throws IOException {
+    //setup HelixParticipant and dependencies
+    String partitionIdStr = "somePartitionId";
+    String partitionIdStr2 = "someOtherPartitionId";
+    ReplicaId replicaId = createMockAmbryReplica(partitionIdStr);
+    ReplicaId replicaId2 = createMockAmbryReplica(partitionIdStr2);
+    String hostname = "localhost";
+    int port = 2200;
+    String instanceName = ClusterMapUtils.getInstanceName(hostname, port);
+    HelixParticipant helixParticipant =
+        new HelixParticipant(new ClusterMapConfig(new VerifiableProperties(props)), helixManagerFactory);
+    helixParticipant.initialize(hostname, port, Collections.EMPTY_LIST);
+    HelixManager helixManager = helixManagerFactory.getZKHelixManager(null, null, null, null);
+    HelixAdmin helixAdmin = helixManager.getClusterManagmentTool();
+    InstanceConfig instanceConfig = new InstanceConfig("someInstanceId");
+    helixAdmin.setInstanceConfig(clusterName, instanceName, instanceConfig);
+    instanceConfig = helixAdmin.getInstanceConfig(clusterName, instanceName);
+    InstanceConfig ic = helixAdmin.getInstanceConfig(clusterName, instanceName);
+
+    //Make sure the current sealedReplicas list is null
+    List<String> sealedReplicas = ClusterMapUtils.getSealedReplicas(instanceConfig);
+    assertNull("sealedReplicas is not null", sealedReplicas);
+
+    String listName = "sealedReplicas";
+
+    //Check that invoking setReplicaSealedState with a non-AmbryReplica ReplicaId throws an IllegalArgumentException
+    ReplicaId notAmbryReplica = createMockNotAmbryReplica(partitionIdStr);
+    try {
+      helixParticipant.setReplicaSealedState(notAmbryReplica, true);
+      fail("Expected an IllegalArgumentException here");
+    }
+    catch (Exception e) {
```

Comment:
not formatted

Reply:
also just catch `IllegalArgumentException`. Any other exception will fail the test anyway